Viper.xctemplate/ViewFileHeaderComments/___FILEBASENAME___Router.swift

```diff
@@ -20,9 +20,11 @@ final class ___VARIABLE_MODULENAME___Router {
     private let dependencies: ___VARIABLE_MODULENAME___RouterDependenciesProtocol
     private weak var view: UIView?
 
-    init(dependencies: ___VARIABLE_MODULENAME___RouterDependenciesProtocol,
-    	 view: UIView?) {
+    init(dependencies: ___VARIABLE_MODULENAME___RouterDependenciesProtocol) {
         self.dependencies = dependencies
+    }
+    
+    func setView(_ view: UIView) {
```

Comment:
Hi Christian, what's the benefit of using a method(setView) here instead of the initializer? 
Did you do this to bring unity to the structure? 
I'm fine with it but I would like to here your main-intention driving this thing forward.

Reply:
Hi Sebastian,

the benefit is that the _presenter reference in the view is now_ **non optional**.

To accomplish this we have to create the components in a specific order:

1. Router
2. Presenter
3. View

Because the view is created last you don't have a view reference at the time of the router and presenter creation.